js/app.js

```diff
@@ -0,0 +1,454 @@
+
+/*---------------------------- Variables (state) ----------------------------*/
+//possible values 0 is covered no bomb, -1 covered bomb, 1-8 neighber bombs
+let gameBoard = []
+let revealed = []
+let mineLocations = []
+let winner
+let numMines = 10
+
+
+
+
+
+/*------------------------ Cached Element References ------------------------*/
+
+
+let boardEl = document.querySelector('.container')
+let allSquares 
+let messageEl = document.getElementById('message')
+let coveredSquares = document.querySelectorAll('.covered')
+let safeSquaresEl = document.querySelectorAll('.safe')
+let replayButton = document.getElementById('play-again')
+let pictureDiv = document.getElementById('picture-div')
```

Comment:
You should be declaring most of your cached element references as `const`s to prevent re-assignment.